tests/test_zone_translation.py

```diff
@@ -0,0 +1,382 @@
+    ----------
+    weighted_config:
+        weighted config to modify and use for a new translation.
+    Returns
+    -------
+        A new config with lower zoning and zone 2 the same.
+    """
+    config = weighted_config.copy()
+    config.zone_2 = weighted_config.lower_zoning
+    config.zone_2.name = weighted_config.zone_2.name
+    config.lower_zoning.name = weighted_config.zone_2.name
+    return config
+
+
+@pytest.fixture(scope="class")
+def expected_weighted() -> pd.DataFrame:
+    """
+    The expected output from a weighted translation using vanila inputs.
+    Compare to output from weighted translation in testing
+    Returns:
+        pd.DataFrame: 4 columns of zone_1, zone_2, zone_1_to_zone_2,
+        zone_2_to_zone_1
+    """
+    output = pd.DataFrame(
+        {
+            "zone_1_id": ["A", "A", "A", "A", "B", "B", "C", "C"],
+            "zone_2_id": ["Z", "X", "Y", "W", "Z", "X", "Z", "Y"],
+            "zone_1_to_zone_2": [
+                0.059,
+                0.176,
+                0.235,
+                0.529,
+                0.263,
+                0.737,
+                0.500,
+                0.500,
+            ],
+            "zone_2_to_zone_1": [
+                0.053,
+                0.176,
+                0.235,
+                1.000,
+                0.263,
+                0.824,
+                0.684,
+                0.765,
+            ],
+        }
+    )
+    return output
+
+
+@pytest.fixture(scope="class")
+def expected_spatial() -> pd.DataFrame:
+    """
+    The expected output from a weighted translation using vanila inputs.
+    Compare to output from weighted translation in testing
+    Returns:
+        pd.DataFrame: 4 columns of zone_1, zone_2, zone_1_to_zone_2,
+        zone_2_to_zone_1
+    """
+    output = pd.DataFrame(
+        {
+            "zone_1_id": ["A", "A", "A", "A", "B", "B", "C", "C"],
+            "zone_2_id": ["Z", "X", "Y", "W", "Z", "X", "Z", "Y"],
+            "zone_1_to_zone_2": [
+                0.05,
+                0.2,
+                0.15,
+                0.6,
+                0.2,
+                0.8,
+                0.625,
+                0.375,
+            ],
+            "zone_2_to_zone_1": [
+                0.05,
+                0.2,
+                0.25,
+                1.000,
+                0.2,
+                0.8,
+                0.75,
+                0.75,
+            ],
+        }
+    )
+    return output
+
+
+@pytest.fixture(scope="class")
+def weighted_trans(weighted_config):
+    trans = zone_translation.ZoneTranslation(weighted_config).zone_translation
+    return trans
+
+
+@pytest.fixture(scope="class")
+def spatial_trans(spatial_config):
+    trans = zone_translation.ZoneTranslation(spatial_config).zone_translation
+    return trans
+
+@pytest.fixture(scope="class")
+def dupe_trans(dupe_shapes_config):
+    trans = zone_translation.ZoneTranslation(dupe_shapes_config).zone_translation
+    return trans
```

Comment:
zone_translation is the actual final product of the tool. I've made these fixtures in case they're needed for more tests but as is they don't need to be, no

Reply:
I would argue that it makes more sense in the tests to just run and extract this data in each test.

We're not having much impact on speed as the translation is so small, so readability trumps speed here I think!